### PR TITLE
SC-31 # Added call to service for deployment

### PR DESCRIPTION
**Pull request by mymattcarroll**
### Added

- SC-20: Call to deployment service when running `bm server deploy`

### Code Review Notes

- Known incomplete work to be tackled in future:

  - [ ] Update information logged to console after a successful deployment with base url of deployed routes/apis.

### CODE REVIEW

File: lib/deploy.js [hunk 1]
```diff
@@ -74,22 +77,40 @@ Please check configuration before continuing
...
+  const stopUpdates = logUpdates(() => `Deploying project - this may take several minutes...`)
+  return new Promise((resolve, reject) => {
+    request({
+      auth: {
+        bearer: accessToken
+      },
+      json: {
+        bundleBucket: serviceSettings.bucket,
+        bundleKey,
+        targetOrigin: 'https://example.com' // TEMP, needed for authorisation
```
Comment (jokeyrhyme) [inline lib/deploy.js:94]: I hadn't picked up on this before, but most of the tooling I've seen uses `// TODO` or a few other keywords for this sort of reminder: http://eslint.org/docs/developer-guide/code-conventions#comment-annotations

`// TEMP` is not incorrect or anything, so don't bother changing this now. Just something to think about and discuss as a group. Maybe TEMP is better? /shrug

File: test/deploy.js [hunk 2]
```diff
@@ -75,11 +78,15 @@ test('authenticate() should call blinkMobileIdentity functions and stop updates'
...
+    },
+    getAccessToken: () => {
+      t.pass()
+      return Promise.resolve()
     }
   })
 })
 
-test('authenticate() should call log correct updates if blinkMobileIdentity functions throw errors', (t) => {
+test.only('authenticate() should call log correct updates if blinkMobileIdentity functions throw errors', (t) => {
```
Comment (jokeyrhyme) [inline test/deploy.js]: You probably didn't mean to check this in with `.only` :)

File: lib/deploy.js (hunk 1 above)
Comment (mymattcarroll) [reply lib/deploy.js:94]: Yeah, I usually use `TODO` but this felt more like it would actually get removed by the end of day. meh

**Comment by jokeyrhyme** [comment]
LGTM 👍

**Pull request merged**